src/main/scala/reporting/Converter.scala

```diff
@@ -0,0 +1,531 @@
+    term match {
+      case Unit              => UnprocessedModelEntry(ConstantEntry("$Snap.unit"))
+      case IntLiteral(x)     => LitIntEntry(x)
+      case t: BooleanLiteral => LitBoolEntry(t.value)
+      case Null()            => VarEntry(model.entries("$Ref.null").toString, sorts.Ref)
+      case Var(_, sort) =>
+        val key: String = term.toString
+        val entry: Option[ModelEntry] = model.entries.get(key)
+        entry
+          .map(x => getConstantEntry(sort, x))
+          .getOrElse(
+            OtherEntry(s"$term", "variable not found in model")
+          )
+
+      case App(app, args) =>
+        /* not tested yet, not sure for which examples this occurs on heap*/
+        var fname = s"${app.id}%limited"
+        if (!model.entries.contains(fname)) {
+          fname = app.id.toString
+          if (!model.entries.contains(fname)) {
+            fname = fname.replace("[", "<").replace("]", ">")
+          }
+        }
+        val toSort = app.resultSort
+        val argEntries: Seq[ExtractedModelEntry] = args
+          .map(t => evaluateTerm(t, model))
+
+        val argsFinal = argEntries.map{
+          case UnprocessedModelEntry(entry) => entry
+          case e: ExtractedModelEntry => ConstantEntry(e.toString)
+        }
+        getFunctionValue(model, fname, argsFinal, toSort)
+
+      case Combine(p0, p1) =>
+        //assuming Combine can only contain other snap.combine and snap.unit
+        val p0Eval = evaluateTerm(p0, model).asValueEntry
+        val p1Eval = evaluateTerm(p1, model).asValueEntry
+        val entry = ApplicationEntry("$Snap.combine", Seq(p0Eval, p1Eval))
+        UnprocessedModelEntry(entry)
+
+      case First(p) =>
+        val sub = evaluateTerm(p, model)
+        sub match {
+          case UnprocessedModelEntry(ApplicationEntry(name, args)) =>
+            if (name == "$Snap.combine") {
+              UnprocessedModelEntry(args.head)
+            } else {
+              OtherEntry(s"First($p)", "unapplicable")
+            }
+          case OtherEntry(t, _) => OtherEntry(s"First($t)", "unapplicable")
+          case _                => OtherEntry(s"First($sub)", "unapplicable")
+        }
+
+      case Second(p) =>
+        val sub = evaluateTerm(p, model)
+        sub match {
+          case UnprocessedModelEntry(ApplicationEntry(name, args)) =>
+            if (name == "$Snap.combine") {
+              UnprocessedModelEntry(args(1))
+            } else {
+              OtherEntry(s"Second($p})", "unapplicable")
+            }
+          case OtherEntry(t, _) => OtherEntry(s"Second($t)", "unapplicable")
+          case _                => OtherEntry(s"Second($sub)", "unapplicable")
+        }
+
+      case SortWrapper(t, to) =>
+        val sub = evaluateTerm(t, model)
+        val fromSortName: String = translateSort(t.sort)
+        val toSortName: String = translateSort(to)
+        val fname = s"$$SortWrappers.${fromSortName}To$toSortName"
```

Comment:
Same comment as above: repeating such string literals here makes the overall code brittle.